db/migrate/20210327053454_add_email_authentication.rb

```diff
@@ -0,0 +1,11 @@
+class AddEmailAuthentication < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :confirmation_token, :string
+    add_column :users, :confirmed_at, :datetime
+    add_column :users, :confirmation_sent_at, :datetime
+    add_column :users, :failed_attempts, :integer, :null => false, :default => 0
+    add_column :users, :unlock_token, :string
+    add_column :users, :locked_at, :datetime
+    add_column :users, :unconfirmed_email, :string
```

Comment:
Does devise natively handle this via functionality via database attributes? I imagine they'd have some sort of caching or temporary local filedump for this kind of data, but if this is how devise handles this stuff, then that's alright.

Reply:
devise handles this, but the columns need to be present to work